### ISSUE TITLE
On alphafold v2.2, but unable to to run neural network portion

### ISSUE THREAD

**Issue by vhu43** [deepmind/alphafold#450]
We were trying to fold some proteins and we previously had a working version of alphafold, but wanted to go to V2.2 for unified memory.

We are running with a ryzen 3950, Nvidia 3060 with 12GB GRam, and 128 ECC ram.  

I tried testing this, but I am encountering an issue, where I'm unable to initialized backend 'tpu driver'.

Below is a copy of the error log
`I0428 15:01:36.602266 140111962661760 run_docker.py:113] Mounting /home/redwood/miniconda3/envs/alphafold/alphafold -> /mnt/fasta_path_0
I0428 15:01:36.602380 140111962661760 run_docker.py:113] Mounting /home/redwood/10TB_HDD_internal/alphafold/alphafold_databases/uniref90 -> /mnt/uniref90_database_path
I0428 15:01:36.602448 140111962661760 run_docker.py:113] Mounting /home/redwood/10TB_HDD_internal/alphafold/alphafold_databases/mgnify -> /mnt/mgnify_database_path
I0428 15:01:36.602503 140111962661760 run_docker.py:113] Mounting /home/redwood/10TB_HDD_internal/alphafold/alphafold_databases -> /mnt/data_dir
I0428 15:01:36.602561 140111962661760 run_docker.py:113] Mounting /home/redwood/10TB_HDD_internal/alphafold/alphafold_databases/pdb_mmcif/mmcif_files -> /mnt/template_mmcif_dir
I0428 15:01:36.602621 140111962661760 run_docker.py:113] Mounting /home/redwood/10TB_HDD_internal/alphafold/alphafold_databases/pdb_mmcif -> /mnt/obsolete_pdbs_path
I0428 15:01:36.602684 140111962661760 run_docker.py:113] Mounting /home/redwood/10TB_HDD_internal/alphafold/alphafold_databases/pdb70 -> /mnt/pdb70_database_path
I0428 15:01:36.602745 140111962661760 run_docker.py:113] Mounting /home/redwood/10TB_HDD_internal/alphafold/alphafold_databases/uniclust30/uniclust30_2018_08 -> /mnt/uniclust30_database_path
I0428 15:01:36.602808 140111962661760 run_docker.py:113] Mounting /home/redwood/10TB_HDD_internal/alphafold/alphafold_databases/bfd -> /mnt/bfd_database_path
I0428 15:01:40.021250 140111962661760 run_docker.py:255] I0428 22:01:40.020589 140496330504000 templates.py:857] Using precomputed obsolete pdbs /mnt/obsolete_pdbs_path/obsolete.dat.
I0428 15:01:42.430200 140111962661760 run_docker.py:255] I0428 22:01:42.429710 140496330504000 tpu_client.py:54] Starting the local TPU driver.
I0428 15:01:42.433193 140111962661760 run_docker.py:255] I0428 22:01:42.432918 140496330504000 xla_bridge.py:212] Unable to initialize backend 'tpu_driver': Not found: Unable to find driver in registry given worker: local://
I0428 15:01:42.512510 140111962661760 run_docker.py:255] I0428 22:01:42.512047 140496330504000 xla_bridge.py:212] Unable to initialize backend 'tpu': Invalid argument: TpuPlatform is not available.
I0428 15:01:52.176146 140111962661760 run_docker.py:255] I0428 22:01:52.175272 140496330504000 run_alphafold.py:377] Have 5 models: ['model_1_pred_0', 'model_2_pred_0', 'model_3_pred_0', 'model_4_pred_0', 'model_5_pred_0']
I0428 15:01:52.176414 140111962661760 run_docker.py:255] I0428 22:01:52.175428 140496330504000 run_alphafold.py:393] Using random seed 1153813040215190560 for the data pipeline
I0428 15:01:52.176526 140111962661760 run_docker.py:255] I0428 22:01:52.175594 140496330504000 run_alphafold.py:161] Predicting T1050
I0428 15:01:52.176642 140111962661760 run_docker.py:255] I0428 22:01:52.176115 140496330504000 jackhmmer.py:133] Launching subprocess "/usr/bin/jackhmmer -o /dev/null -A /tmp/tmp_znpkqst/output.sto --noali --F1 0.0005 --F2 5e-05 --F3 5e-07 --incE 0.0001 -E 0.0001 --cpu 8 -N 1 /mnt/fasta_path_0/T1050.fasta /mnt/uniref90_database_path/uniref90.fasta"
I0428 15:01:52.346133 140111962661760 run_docker.py:255] I0428 22:01:52.345630 140496330504000 utils.py:36] Started Jackhmmer (uniref90.fasta) query
I0428 15:08:27.362812 140111962661760 run_docker.py:255] I0428 22:08:27.362295 140496330504000 utils.py:40] Finished Jackhmmer (uniref90.fasta) query in 395.016 seconds
I0428 15:08:31.379740 140111962661760 run_docker.py:255] I0428 22:08:31.377884 140496330504000 jackhmmer.py:133] Launching subprocess "/usr/bin/jackhmmer -o /dev/null -A /tmp/tmpioxnqda3/output.sto --noali --F1 0.0005 --F2 5e-05 --F3 5e-07 --incE 0.0001 -E 0.0001 --cpu 8 -N 1 /mnt/fasta_path_0/T1050.fasta /mnt/mgnify_database_path/mgy_clusters_2018_12.fa"
I0428 15:08:31.556135 140111962661760 run_docker.py:255] I0428 22:08:31.555570 140496330504000 utils.py:36] Started Jackhmmer (mgy_clusters_2018_12.fa) query
I0428 15:14:45.565228 140111962661760 run_docker.py:255] I0428 22:14:45.564686 140496330504000 utils.py:40] Finished Jackhmmer (mgy_clusters_2018_12.fa) query in 374.009 seconds
I0428 15:15:02.867276 140111962661760 run_docker.py:255] I0428 22:15:02.866789 140496330504000 hhsearch.py:85] Launching subprocess "/usr/bin/hhsearch -i /tmp/tmplhjpmjxb/query.a3m -o /tmp/tmplhjpmjxb/output.hhr -maxseq 1000000 -d /mnt/pdb70_database_path/pdb70"
I0428 15:15:03.043247 140111962661760 run_docker.py:255] I0428 22:15:03.042704 140496330504000 utils.py:36] Started HHsearch query
I0428 15:16:47.714468 140111962661760 run_docker.py:255] I0428 22:16:47.713911 140496330504000 utils.py:40] Finished HHsearch query in 104.671 seconds
I0428 15:16:53.952752 140111962661760 run_docker.py:255] I0428 22:16:53.952196 140496330504000 hhblits.py:128] Launching subprocess "/usr/bin/hhblits -i /mnt/fasta_path_0/T1050.fasta -cpu 4 -oa3m /tmp/tmpwbd6nz4m/output.a3m -o /dev/null -n 3 -e 0.001 -maxseq 1000000 -realign_max 100000 -maxfilt 100000 -min_prefilter_hits 1000 -d /mnt/bfd_database_path/bfd_metaclust_clu_complete_id30_c90_final_seq.sorted_opt -d /mnt/uniclust30_database_path/uniclust30_2018_08"
I0428 15:16:54.133208 140111962661760 run_docker.py:255] I0428 22:16:54.132631 140496330504000 utils.py:36] Started HHblits query
I0428 16:28:19.232026 140111962661760 run_docker.py:255] I0428 23:28:19.230983 140496330504000 utils.py:40] Finished HHblits query in 4285.098 seconds
I0428 16:28:19.728557 140111962661760 run_docker.py:255] I0428 23:28:19.728026 140496330504000 templates.py:878] Searching for template for: MASQSYLFKHLEVSDGLSNNSVNTIYKDRDGFMWFGTTTGLNRYDGYTFKIYQHAENEPGSLPDNYITDIVEMPDGRFWINTARGYVLFDKERDYFITDVTGFMKNLESWGVPEQVFVDREGNTWLSVAGEGCYRYKEGGKRLFFSYTEHSLPEYGVTQMAECSDGILLIYNTGLLVCLDRATLAIKWQSDEIKKYIPGGKTIELSLFVDRDNCIWAYSLMGIWAYDCGTKSWRTDLTGIWSSRPDVIIHAVAQDIEGRIWVGKDYDGIDVLEKETGKVTSLVAHDDNGRSLPHNTIYDLYADRDGVMWVGTYKKGVSYYSESIFKFNMYEWGDITCIEQADEDRLWLGTNDHGILLWNRSTGKAEPFWRDAEGQLPNPVVSMLKSKDGKLWVGTFNGGLYCMNGSQVRSYKEGTGNALASNNVWALVEDDKGRIWIASLGGGLQCLEPLSGTFETYTSNNSALLENNVTSLCWVDDNTLFFGTASQGVGTMDMRTREIKKIQGQSDSMKLSNDAVNHVYKDSRGLVWIATREGLNVYDTRRHMFLDLFPVVEAKGNFIAAITEDQERNMWVSTSRKVIRVTVASDGKGSYLFDSRAYNSEDGLQNCDFNQRSIKTLHNGIIAIGGLYGVNIFAPDHIRYNKMLPNVMFTGLSLFDEAVKVGQSYGGRVLIEKELNDVENVEFDYKQNIFSVSFASDNYNLPEKTQYMYKLEGFNNDWLTLPVGVHNVTFTNLAPGKYVLRVKAINSDGYVGIKEATLGIVVNPPFKLAAALQHHHHHH
I0428 16:28:21.533290 140111962661760 run_docker.py:255] I0428 23:28:21.532789 140496330504000 templates.py:268] Found an exact template match 4a2m_B.
I0428 16:28:24.055201 140111962661760 run_docker.py:255] I0428 23:28:24.054719 140496330504000 templates.py:268] Found an exact template match 4a2l_F.
I0428 16:28:25.840406 140111962661760 run_docker.py:255] I0428 23:28:25.839844 140496330504000 templates.py:268] Found an exact template match 3v9f_B.
I0428 16:28:26.944004 140111962661760 run_docker.py:255] I0428 23:28:26.943541 140496330504000 templates.py:268] Found an exact template match 3va6_A.
I0428 16:28:27.768368 140111962661760 run_docker.py:255] I0428 23:28:27.767889 140496330504000 templates.py:268] Found an exact template match 3ott_B.
I0428 16:28:28.192252 140111962661760 run_docker.py:255] I0428 23:28:28.191807 140496330504000 templates.py:268] Found an exact template match 5m11_A.
I0428 16:28:28.220198 140111962661760 run_docker.py:255] I0428 23:28:28.219779 140496330504000 templates.py:268] Found an exact template match 4a2m_B.
I0428 16:28:28.248700 140111962661760 run_docker.py:255] I0428 23:28:28.248321 140496330504000 templates.py:268] Found an exact template match 4a2l_F.
I0428 16:28:28.277440 140111962661760 run_docker.py:255] I0428 23:28:28.277111 140496330504000 templates.py:268] Found an exact template match 4a2m_B.
I0428 16:28:28.305952 140111962661760 run_docker.py:255] I0428 23:28:28.305578 140496330504000 templates.py:268] Found an exact template match 5m11_A.
I0428 16:28:28.333765 140111962661760 run_docker.py:255] I0428 23:28:28.333357 140496330504000 templates.py:268] Found an exact template match 4a2l_F.
I0428 16:28:28.362686 140111962661760 run_docker.py:255] I0428 23:28:28.362404 140496330504000 templates.py:268] Found an exact template match 3v9f_B.
I0428 16:28:28.391016 140111962661760 run_docker.py:255] I0428 23:28:28.390779 140496330504000 templates.py:268] Found an exact template match 3ott_B.
I0428 16:28:28.419734 140111962661760 run_docker.py:255] I0428 23:28:28.419532 140496330504000 templates.py:268] Found an exact template match 3va6_A.
I0428 16:28:28.448667 140111962661760 run_docker.py:255] I0428 23:28:28.448436 140496330504000 templates.py:268] Found an exact template match 3ott_B.
I0428 16:28:28.477515 140111962661760 run_docker.py:255] I0428 23:28:28.477142 140496330504000 templates.py:268] Found an exact template match 3va6_A.
I0428 16:28:28.506501 140111962661760 run_docker.py:255] I0428 23:28:28.506254 140496330504000 templates.py:268] Found an exact template match 5m11_A.
I0428 16:28:28.534357 140111962661760 run_docker.py:255] I0428 23:28:28.534104 140496330504000 templates.py:268] Found an exact template match 4a2m_B.
I0428 16:28:28.563012 140111962661760 run_docker.py:255] I0428 23:28:28.562647 140496330504000 templates.py:268] Found an exact template match 4a2l_F.
I0428 16:28:28.591441 140111962661760 run_docker.py:255] I0428 23:28:28.591219 140496330504000 templates.py:268] Found an exact template match 3v9f_B.
I0428 16:28:30.618727 140111962661760 run_docker.py:255] I0428 23:28:30.618245 140496330504000 pipeline.py:234] Uniref90 MSA size: 10000 sequences.
I0428 16:28:30.618900 140111962661760 run_docker.py:255] I0428 23:28:30.618381 140496330504000 pipeline.py:235] BFD MSA size: 4982 sequences.
I0428 16:28:30.618991 140111962661760 run_docker.py:255] I0428 23:28:30.618414 140496330504000 pipeline.py:236] MGnify MSA size: 501 sequences.
I0428 16:28:30.619096 140111962661760 run_docker.py:255] I0428 23:28:30.618446 140496330504000 pipeline.py:238] Final (deduplicated) MSA size: 15419 sequences.
I0428 16:28:30.619187 140111962661760 run_docker.py:255] I0428 23:28:30.618657 140496330504000 pipeline.py:241] Total number of templates (NB: this can include bad templates and is later filtered to top 4): 20.
I0428 16:28:30.749678 140111962661760 run_docker.py:255] I0428 23:28:30.742394 140496330504000 run_alphafold.py:190] Running model model_1_pred_0 on T1050
I0428 16:28:39.006426 140111962661760 run_docker.py:255] I0428 23:28:39.005879 140496330504000 model.py:166] Running predict with shape(feat) = {'aatype': (4, 779), 'residue_index': (4, 779), 'seq_length': (4,), 'template_aatype': (4, 4, 779), 'template_all_atom_masks': (4, 4, 779, 37), 'template_all_atom_positions': (4, 4, 779, 37, 3), 'template_sum_probs': (4, 4, 1), 'is_distillation': (4,), 'seq_mask': (4, 779), 'msa_mask': (4, 508, 779), 'msa_row_mask': (4, 508), 'random_crop_to_size_seed': (4, 2), 'template_mask': (4, 4), 'template_pseudo_beta': (4, 4, 779, 3), 'template_pseudo_beta_mask': (4, 4, 779), 'atom14_atom_exists': (4, 779, 14), 'residx_atom14_to_atom37': (4, 779, 14), 'residx_atom37_to_atom14': (4, 779, 37), 'atom37_atom_exists': (4, 779, 37), 'extra_msa': (4, 5120, 779), 'extra_msa_mask': (4, 5120, 779), 'extra_msa_row_mask': (4, 5120), 'bert_mask': (4, 508, 779), 'true_msa': (4, 508, 779), 'extra_has_deletion': (4, 5120, 779), 'extra_deletion_value': (4, 5120, 779), 'msa_feat': (4, 508, 779, 49), 'target_feat': (4, 779, 22)}
I0428 16:29:33.725177 140111962661760 run_docker.py:255] 2022-04-28 23:29:33.723981: E external/org_tensorflow/tensorflow/stream_executor/cuda/cuda_blas.cc:226] failed to create cublas handle: CUBLAS_STATUS_NOT_INITIALIZED
I0428 16:29:33.725364 140111962661760 run_docker.py:255] 2022-04-28 23:29:33.724034: F external/org_tensorflow/tensorflow/compiler/xla/service/gpu/gemm_algorithm_picker.cc:113] Check failed: stream->parent()->GetBlasGemmAlgorithms(&algorithms)
I0428 16:29:33.725598 140111962661760 run_docker.py:255] Fatal Python error: Aborted
I0428 16:29:33.725669 140111962661760 run_docker.py:255] 
I0428 16:29:33.725742 140111962661760 run_docker.py:255] Current thread 0x00007fc7d9df3740 (most recent call first):
I0428 16:29:33.725801 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/xla.py", line 360 in backend_compile
I0428 16:29:33.725855 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/xla.py", line 743 in _xla_callable
I0428 16:29:33.725905 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/linear_util.py", line 262 in memoized_fun
I0428 16:29:33.725954 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/xla.py", line 593 in _xla_call_impl
I0428 16:29:33.726002 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/core.py", line 606 in process_call
I0428 16:29:33.726051 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/core.py", line 1563 in process
I0428 16:29:33.726101 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/core.py", line 1551 in call_bind
I0428 16:29:33.726156 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/core.py", line 1560 in bind
I0428 16:29:33.726207 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/_src/api.py", line 427 in cache_miss
I0428 16:29:33.726263 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/jax/_src/traceback_util.py", line 183 in reraise_with_filtered_traceback
I0428 16:29:33.726308 140111962661760 run_docker.py:255] File "/app/alphafold/alphafold/model/model.py", line 167 in predict
I0428 16:29:33.726349 140111962661760 run_docker.py:255] File "/app/alphafold/run_alphafold.py", line 199 in predict_structure
I0428 16:29:33.726391 140111962661760 run_docker.py:255] File "/app/alphafold/run_alphafold.py", line 406 in main
I0428 16:29:33.726434 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/absl/app.py", line 258 in _run_main
I0428 16:29:33.726479 140111962661760 run_docker.py:255] File "/opt/conda/lib/python3.7/site-packages/absl/app.py", line 312 in run
I0428 16:29:33.726521 140111962661760 run_docker.py:255] File "/app/alphafold/run_alphafold.py", line 422 in <module>
I0428 16:29:36.101321 140111962661760 run_docker.py:255] /app/run_alphafold.sh: line 3:     8 Aborted                 (core dumped) python /app/alphafold/run_alphafold.py "$@"
`

**Comment by vhu43** [comment]
We were able to GPU fold proteins with the release version, and we can detect the gpu with docker.

**Comment by vhu43** [comment]
Updated cudNN and this has been resolved.
